SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -711,27 +711,10 @@ void ActorSystem::HandleIndividualAddComponent(const Worker_EntityId EntityId, c
 	}
 
 	// Otherwise this is a dynamically attached component. We need to make sure we have all related components before creation.
-	PendingDynamicSubobjectComponents.Add(MakeTuple(static_cast<Worker_EntityId_Key>(EntityId), ComponentId));
-
-	bool bReadyToCreate = true;
-	ForAllSchemaComponentTypes([&](ESchemaComponentType Type) {
-		Worker_ComponentId SchemaComponentId = Info.SchemaComponents[Type];
+	PendingDynamicSubobjectComponents.FindOrAdd(EntityId).Emplace(ComponentId);
 
-		// TODO (UNR-4460): This should not require checking out all the subobject components, only the ones
-		// we could check out. If data is always present, we could send that last and attach subobjects only when we
-		// see data.
-		if (SchemaComponentId == SpatialConstants::INVALID_COMPONENT_ID)
-		{
-			return;
-		}
-
-		if (!PendingDynamicSubobjectComponents.Contains(MakeTuple(static_cast<Worker_EntityId_Key>(EntityId), SchemaComponentId)))
-		{
-			bReadyToCreate = false;
-		}
-	});
-
-	if (bReadyToCreate)
+	// We ensure that the data component is added last, so once it's in view we're ready to attach the Unreal subobject
+	if (ComponentId == Info.SchemaComponents[SCHEMA_Data])
```

Comment:
Use specific **SCHEMA_Data** here means we do not support **SCHEMA_OwnerOnly** any more. 
Is this will be a problem here?

Reply:
We only check for the Data component here because as we attach it last we know that all other relevant spatial components are in view at this point. We know then that the Unreal component is ready to be created.

Reply:
Is that a guarantee from the runtime and worker SDK? That we will always receive the components in the order we have written them?

Reply:
I thought it was - but this turned out to not be the case - this functionality has been re-written.